src/lib.rs

```diff
@@ -0,0 +1,102 @@
+#[allow(dead_code)]
+#[derive(Debug)]
+#[derive(PartialEq)]
+pub enum StatusState {
+    /// Works right out of the box without any preperation steps needed
+    OK,
+    /// Works, but preparation steps are needed
+    PREPARATION,
+    /// Doesn't work (too unstable to use falls also in this category)
+    BROKEN,
+}
+#[allow(dead_code)]
+#[derive(Debug)]
+#[derive(PartialEq)]
+/// The status of a provider
+pub struct Status {
+    pub state: StatusState,
+    /// Date of when the state was last checked/updated
+    pub date: &'static str,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+#[derive(PartialEq)]
+/// Information about a provider
+pub struct Provider {
+    /// for linking to the providers page on the overview website
+    /// like `providers.delta.chat/{overview_page}`
+    pub overview_page: &'static str,
+    pub name: &'static str,
+    pub status: Status,
+    /// The markdown content of the providers page containing the preparation steps
+    pub markdown: &'static str,
+}
+
+#[allow(dead_code)]
+#[derive(Debug)]
+#[derive(PartialEq)]
+struct DomainDBEntry {
+    domain: &'static str,
+    list_index: u32,
+}
+
+include!(concat!(env!("OUT_DIR"), "/data.rs"));
+
+/// Get the domain part of an valid email address
+pub fn get_domain_from_email(valid_email_address: &str) -> &str {
```

Comment:
Consider returning `Result`?  There's no guarantee that the user will provide an email address in a valid form and panicking in that case seems undesired.

Also, and very much optional, dcc-rs uses the `addr` crate to parse email addresses.  If you like you could try use that here too for more correctness.  Though you'd probably get a long way with this simplified version.

Reply:
Thinking about this a bit more, maybe this isn't even a function this crate should be providing at all?

Reply:
Core already checks the email address, then it has to pass only valid adresses. 

The only case in which this would panic is when there is no "@" in the string, and it is unlikely that core could miss such an obvious problem.

Reply:
Actually, I was wrong. It will not panic at all.